lucene/expressions/src/test/org/apache/lucene/expressions/TestExpressionValueSource.java

```diff
@@ -128,6 +128,34 @@ public void testDoubleValuesSourceEquals() throws Exception {
     assertFalse(vs1.equals(vs4));
   }
 
+  public void testFibonacciExpr() throws Exception {
+    int n = 40;
+    SimpleBindings bindings = new SimpleBindings();
+    bindings.add("f0", DoubleValuesSource.constant(0));
+    bindings.add("f1", DoubleValuesSource.constant(1));
+    for (int i = 2; i < n + 1; i++) {
```

Comment:
Maybe adds comments to explain that before the fix, this test takes XXX hours to finish, and after the fix it's nearly instantaneous as judged by casual human observer?

Reply:
Sure.